### PR TITLE
Try find config in `~/.config` or `$XDG_CONFIG_HOME`

### PR DESCRIPTION
**Pull request by vladsavelyev**
Fix https://github.com/ewels/MultiQC/issues/1682

### CODE REVIEW

File: docs/core/getting_started/config.md [hunk 1]
```diff
@@ -13,15 +13,16 @@ it collects the configuration settings from the following places in this order
 1. Hardcoded defaults in MultiQC code
 2. System-wide config in `<installation_dir>/multiqc_config.yaml`
    - Manual installations only, not `pip` or `conda`
-3. User config in `~/.multiqc_config.yaml`
-4. File path set in environment variable `MULTIQC_CONFIG_PATH`
+3. User config in `$XDG_CONFIG_HOME/multiqc_config.yaml` (or `~/.config/.multiqc_config.yaml` if `$XDG_CONFIG_HOME` is not set)
```
Comment (ewels) [inline docs/core/getting_started/config.md:16]:
```suggestion
3. User config in `$XDG_CONFIG_HOME/multiqc_config.yaml` (or `~/.config/multiqc_config.yaml` if `$XDG_CONFIG_HOME` is not set)
```

File: multiqc/utils/config.py [hunk 2]
```diff
@@ -219,6 +219,10 @@ def mqc_load_userconfig(paths=()):
     # Load and parse installation config file if we find it
     mqc_load_config(os.path.join(os.path.dirname(MULTIQC_DIR), "multiqc_config.yaml"))
 
+    # Load and parse a config file in $XDG_CONFIG_HOME
+    # Ref: https://specifications.freedesktop.org/basedir-spec/basedir-spec-latest.html
+    mqc_load_config(os.path.join(os.environ.get("XDG_CONFIG_HOME", "~/.config"), "multiqc_config.yaml"))
```
Comment (ewels) [inline multiqc/utils/config.py:224]: Not sure if we need to expand the user path in the default here?
```suggestion
    mqc_load_config(os.path.join(os.environ.get("XDG_CONFIG_HOME", os.path.expanduser("~/.config")), "multiqc_config.yaml"))
```

**Review by ewels — COMMENTED** [review]
Nice! Two minor comments.

**Comment by vladsavelyev** [comment]
@multiqc-bot fix linting

**Pull request merged**